compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -5209,7 +5214,7 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
           } else if (jsInterop.isJSBracketAccess(sym)) {
             assert(argc == 1 || argc == 2,
                 s"@JSBracketAccess methods should have 1 or 2 non-varargs arguments")
-            argsNoSpread match {
+            (argsNoSpread: @unchecked) match {
```

Comment:
Why not just put a default case throwing the error in the assertion above? That seems less complicated.

Reply:
Ah yes, that's much better of course.